src/accelerate/checkpointing.py

```diff
@@ -139,10 +149,18 @@ def load_accelerator_state(
     logger.info("All model weights loaded successfully")
 
     # Optimizer states
+    if optimizer_map_location not in [None, "cpu", "on_device"]:
+        raise TypeError(
+            "Unsupported optimizer map location passed, please choose one of `None`, `cpu`, or `on_device`"
+        )
+    if optimizer_map_location is None:
+        optimizer_map_location = "cpu"
+    elif optimizer_map_location == "on_device":
```

Comment:
I still think we should default to `"on_device"` for distributed trainings on GPU, otherwise we require num_processes times the optimizer state available in CPU RAM.